hack/patch-nix-hash.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+nix --log-format raw build ./dev/nix/
+if [ $? -eq 0 ]; then
+  echo OK
+else
+  # this is reliant on the vendorHash being on line 195
+  sed -i '195s/vendorHash = \"\([^\"]*\)\"/vendorHash = ""/g' ./dev/nix/flake.nix
```

Comment:
```suggestion
```

we can just replace it in the next `sed` command; no need to run a similar command twice.

Reply:
So this ordering actually matters, this is because the error message for a non present nix hash can be different to the error message for an empty hash.

This means that we can't really get rid of this nor store the output as a variable.

Reply:
But do we actually care to differentiate the error? If there's an error, it needs a manual fix.
I would think we can just do the replace, and if that fails, print a nicer error message that it needs a manual fix and `exit 1`

Reply:
Yeah we do actually care to differentiate the error, only a certain error msg contains the correct nix hash. 
We force the correct error output via the first sed command.

Reply:
I've reduced the if check, and forced this ordering by always doing the first sed.

Reply:
> Yeah we do actually care to differentiate the error, only a certain error msg contains the correct nix hash.

We can just print a nicer error ourselves though rather than relying on `sed`'s error, which is not as user-friendly / specific to the script. Unless I'm misunderstanding something?

The current version seems to also unconditionally print the change even if nothing was changed?

Reply:
Yeah we aren't relying on sed's error. Sed is used to force an ordering which gets the correct error output from the nix build command. 

Yes the current version unconditionally changes the hash, this is the only way to reduce the number of "nix build" commands issued. We can't just store it unfortunately.

Reply:
OH you mean you're intentionally making the `nix build` command fail in order to pull the `vendorHash` from its output??
Oh that is wacky I did not expect that.

> Yes the current version unconditionally changes the hash

I actually didn't mean the hash, I meant the print / `echo`. It always says "Changed" even if it didn't change, which is a bit confusing.

Are you not able to:
1. `grep` current hash
1. `nix build` 
1. `grep` new hash
1. Check if it changed, if so, `echo`

Or is there a reason you can't?